client/src/main/kotlin/io/spine/examples/chatspn/desktop/navigation/NavigationModel.kt

```diff
@@ -0,0 +1,301 @@
+        }
+    }
+
+    /**
+     * Finds user by email and creates the personal chat between authenticated and found user.
+     *
+     * @param email email of the user with whom to create a personal chat
+     */
+    public fun createPersonalChat(email: String) {
+        val user = client.findUser(email)
+        if (null != user) {
+            client.createPersonalChat(user.id)
+        } else {
+            userSearchFieldState.errorState.value = true
+        }
+    }
+
+    /**
+     * Returns the data of the personal chat with the provided user,
+     * or `null` if the chat doesn't exist.
+     *
+     * @param user ID of the user with whom to find the personal chat
+     */
+    private fun findPersonalChat(user: UserId): ChatData? {
+        val chat = chatPreviewsState.value.find { chatData ->
+            chatData.type == Chat.ChatType.CT_PERSONAL && chatData.members.contains(user)
+        }
+        return chat
+    }
+
+    /**
+     * Opens a page with a user profile.
+     *
+     * @param userId id of the user to open profile
+     */
+    public fun openUserProfile(userId: UserId) {
+        val user = client.findUser(userId)
+        profilePageState.userProfile.value = user ?: UserProfile.getDefaultInstance()
+        profilePageState.chatState.value = if (null == user) null else findPersonalChat(user.id)
+        currentPage.value = Page.PROFILE
+    }
+
+    /**
+     * Opens a page with chat info. If the chat is personal, a user profile will be opened.
+     *
+     * @param chatId ID of the chat which info to open
+     */
+    public fun openChatInfo(chatId: ChatId) {
+        val chat = chatPreviewsState.value.find { chatData -> chatId.equals(chatData.id) } ?: return
+        if (chat.type == Chat.ChatType.CT_PERSONAL) {
+            val userId = chat.members.find { user -> !user.equals(authenticatedUser.id) }
+            val user = client.findUser(userId!!)
+            profilePageState.userProfile.value = user!!
+            profilePageState.chatState.value = chatData(chat.id)
+            currentPage.value = Page.PROFILE
+        }
+    }
+}
+
+/**
+ * State of the modal window with the user profile.
+ */
+public class ProfilePageState {
+    public val userProfile: MutableState<UserProfile> =
+        mutableStateOf(UserProfile.getDefaultInstance())
+    public val chatState: MutableState<ChatData?> = mutableStateOf(null)
+
+    /**
+     * Clears the state.
+     */
+    public fun clear() {
+        userProfile.value = UserProfile.getDefaultInstance()
+        chatState.value = null
+    }
+}
+
+/**
+ * State of the user search field.
+ */
+public class UserSearchFieldState {
+    public val userEmailState: MutableState<String> = mutableStateOf("")
+    public val errorState: MutableState<Boolean> = mutableStateOf(false)
+}
+
+/**
+ * Pages in the application.
+ */
+public enum class Page {
+    CHAT, LOGIN, REGISTRATION, PROFILE
+}
+
+/**
+ * Data for the chat preview.
+ */
+public data class ChatData(
```

Comment:
With the documented purpose in mind, this name is too generic, especially for `public` data type.

Reply:
This is still too broad. Please rename more precisely.

And its documentation was fine.